pkg/frontend/transport/handler.go

```diff
@@ -218,7 +218,8 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			}
 			http.Error(w, err.Error(), statusCode)
 			if f.cfg.QueryStatsEnabled && util.IsRequestBodyTooLarge(err) {
-				f.rejectedQueries.WithLabelValues(reasonRequestBodySizeExceeded, userID).Inc()
+				source := tripperware.GetSource(r.Header.Get("User-Agent"))
+				f.rejectedQueries.WithLabelValues(reasonRequestBodySizeExceeded, source, userID).Inc()
```

Comment:
If we allow any different "user-agent" as source, we may have cardinality problem that could cause metrics explosions.

Maybe we could just sub categorize the ruler UA and Others?

Reply:
```
source := tripperware.GetSource(r.Header.Get("User-Agent"))
```

`GetSource` already categorizes it to API and Ruler

Reply:
Oh nice! LGTM